src/screen/show_refs.rs

```diff
@@ -18,57 +22,122 @@ pub(crate) fn create(config: Rc<Config>, repo: Rc<Repository>, size: Rect) -> Re
+            .chain(create_remotes_sections(
+                &repo,
+                &style.section_header,
+                &style.remote,
+            )?)
+            .chain(iter::once(Item {
+                id: "tags".into(),
+                display: Line::styled("Tags".to_string(), &style.section_header),
+                section: true,
+                depth: 0,
+                ..Default::default()
+            }))
+            .chain(create_references_section(
+                &repo,
+                Reference::is_tag,
+                &style.tag,
+            )?)
             .collect())
         }),
     )
 }
+
+fn create_remotes_sections<'a>(
+    repo: &'a Repository,
+    header_style: &'a StyleConfigEntry,
+    item_style: &'a StyleConfigEntry,
+) -> Res<impl Iterator<Item = Item> + 'a> {
+    let all_remotes = create_references_section(repo, Reference::is_remote, item_style)?;
+    let mut remotes = HashMap::new();
+    for remote in all_remotes {
+        let name = match remote.id.split_once('/') {
+            None => remote.id.as_ref(),
+            Some((name, _)) => name,
+        };
+
+        match remotes.entry(name.to_string()) {
+            Entry::Vacant(entry) => {
+                entry.insert(vec![remote]);
+            }
+            Entry::Occupied(mut entry) => {
+                entry.get_mut().push(remote);
+            }
+        }
+    }
+
+    Ok(remotes.into_iter().flat_map(move |(name, items)| {
+        let header = capitalize(&name);
+        iter::once(Item {
+            id: name.into(),
+            display: Line::styled(header, header_style),
+            section: true,
+            depth: 0,
+            ..Default::default()
+        })
+        .chain(items)
+    }))
+}
+
+fn capitalize(word: &str) -> String {
+    let mut chars = word.chars();
+    match chars.next() {
+        Some(c) => c.to_uppercase().collect::<String>() + chars.as_str(),
+        None => String::new(),
+    }
+}
+
+fn create_references_section<'a, F>(
+    repo: &'a Repository,
+    filter: F,
+    style: &'a StyleConfigEntry,
+) -> Res<impl Iterator<Item = Item> + 'a>
+where
+    F: FnMut(&Reference<'a>) -> bool + 'a,
+{
+    Ok(repo
+        .references()?
+        .filter_map(Result::ok)
+        .filter(filter)
+        .map(move |reference| {
+            let name = Span::styled(reference.shorthand().unwrap().to_string(), style);
+
+            Item {
+                id: name.clone().content,
+                display: Line::from(vec![create_prefix(repo, &reference), name.clone()]),
+                depth: 1,
+                target_data: Some(TargetData::Branch(name.content.into())),
+                ..Default::default()
+            }
+        }))
+}
+
+fn create_prefix(repo: &Repository, reference: &Reference) -> Span<'static> {
+    let head = repo.head().ok();
+
+    Span::raw(if repo.head_detached().unwrap_or(false) {
+        if reference.target() == head.as_ref().and_then(Reference::target) {
+            "? "
```

Comment:
I like this! More useful than nothing. :+1: 
In magit it'd just say:
![image](https://github.com/altsem/gitu/assets/3618477/154fa9b6-f0af-4470-b83d-fcf20949ec1f)

Reply:
I glad you liked it :)